src/renderer/components/ImagePlayer.tsx

```diff
@@ -45,25 +49,38 @@ export default class ImagePlayer extends React.Component {
       while (offset < -this.state.historyPaths.length) {
         offset += this.state.historyPaths.length;
       }
-      img = new Image();
+      const img = new Image();
       img.src = this.state.historyPaths[this.state.historyPaths.length + offset];
+      imgs.push(img);
+    } else {
+      const max = this.props.fadeEnabled ? 3 : 2;
+      for (let i=1; i<max; i++) {
+        const img = this.state.pastAndLatest[this.state.pastAndLatest.length - i];
+        if (img) {
+          imgs.push(img);
+        }
+      }
```

Comment:
We now render more than one image at a time, so we can fade out the old image and reveal the new one.